unidist/cli/backends/mpi/runner.py

```diff
@@ -0,0 +1,121 @@
+        self.backend = BackendName.MPI
+        super().__init__(**cli_kwargs)
+
+    def check_kwargs_support(self, **kwargs):
+        """Check support for `kwargs` combination for MPI backend."""
+        hosts = kwargs.get("hosts", Defaults.HOSTS)
+        num_cpus = kwargs.get("num_cpus", Defaults.NUM_CPUS)
+
+        if not isinstance(num_cpus, list):
+            num_cpus = [num_cpus]
+
+        if not isinstance(hosts, list):
+            hosts = [hosts]
+
+        if len(hosts) != len(num_cpus):
+            # If `num_cpus` isn't provided or a single value `default` is provided
+            # all workers will use `default` value
+            if len(num_cpus) == 1 and num_cpus[0] == Defaults.NUM_CPUS:
+                num_cpus *= len(hosts)
+            else:
+                raise RuntimeError(
+                    "`num_cpus` and `hosts` parameters must have the equal number of values."
+                )
+
+        if (
+            kwargs.get("redis_password", Defaults.REDIS_PASSWORD)
+            != Defaults.REDIS_PASSWORD
+        ):
+            warnings.warn(
+                f"`redis_password` isn't supported for {self.backend} backend, ignored.",
+                RuntimeWarning,
+            )
+
+        self.hosts = validate_hosts(hosts)
+        self.num_cpus = validate_num_cpus(num_cpus)
+
+    def prepare_env(self):
+        """Setup unidist environment variables for MPI backend."""
+        super().prepare_env()
+        os.environ["UNIDIST_CPUS"] = str(sum([int(val) for val in self.num_cpus]))
+
+    def get_command(self):
+        """
+        Get a command to be run in a subprocess.
+
+        Returns
+        -------
+        list
+            List of strings with the command representation.
+        """
+        unidist_root = get_unidist_root()
+        workers_dir = "/tmp"
+        command = ["mpiexec", "-hosts"]
+        hosts_str = f"{get_localhost_ip()}:1,{get_localhost_ip()}:1,"
+        for host, n in zip(self.hosts, self.num_cpus):
+            hosts_str += host + ":" + n + ","
+        command_executor = ["-n", "1", self.executor, self.script]
+        command_monitor = [
+            "-n",
+            "1",
+            "-wdir",
+            workers_dir,
+            "python",
+            os.path.join(
+                unidist_root, "unidist", "core", "backends", "mpi", "core", "monitor.py"
+            ),
+        ]
+
+        def get_worker_command(num_cpus):
+            return [
+                "-n",
+                num_cpus,
+                "-wdir",
+                workers_dir,
+                "python",
+                os.path.join(
+                    unidist_root,
+                    "unidist",
+                    "core",
+                    "backends",
+                    "mpi",
+                    "core",
+                    "worker.py",
+                ),
+            ]
+
+        command += [hosts_str] + command_executor + [":"] + command_monitor
```

Comment:
Are you sure `hosts_str` should follow here? Actually, please, provide full commands for running MPI both in a single node and in a cluster (two nodes are enough).

Reply:
https://github.com/modin-project/unidist/pull/14#discussion_r756101033

Reply:
I see, thanks!